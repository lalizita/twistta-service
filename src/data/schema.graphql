input CreateTweetInput {
  author: ID!
  parent: String
  text: String!
  clientMutationId: String
}

type CreateTweetPayload {
  tweets: [Tweet]
  error: String
  clientMutationId: String
}

type Mutation {
  UserRegister(input: RegisterUsernameInput!): RegisterUsernamePayload
  UserLogin(input: UserLoginInput!): UserLoginPayload
  CreateTweet(input: CreateTweetInput!): CreateTweetPayload
}

"""Query of users"""
type Query {
  users: [User]
  tweets(id: ID!): [Tweet]
}

input RegisterUsernameInput {
  username: String!
  email: String!
  password: String!
  clientMutationId: String
}

type RegisterUsernamePayload {
  username: String
  token: String
  clientMutationId: String
}

"""Tweets of a user"""
type Tweet {
  author: ID!
  parent: ID
  text: String
}

"""user data"""
type User {
  _id: String
  username: String!
  active: Boolean
  email: String!
}

input UserLoginInput {
  username: String!
  password: String!
  clientMutationId: String
}

type UserLoginPayload {
  token: String
  error: String
  clientMutationId: String
}
