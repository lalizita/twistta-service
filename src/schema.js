import { GraphQLSchema } from 'graphql';

import QueryType from './type/QueryType';
import MutationType from './type/MutationType';

console.log(MutationType)

export const schema = new GraphQLSchema({
  query: QueryType,
  mutation: MutationType
});