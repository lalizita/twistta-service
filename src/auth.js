import jwt from 'jsonwebtoken';
import UserModel from './models/User';

export async function getUser(token) {
  if (!token) return { user: null };

  try {
    const decodedToken = jwt.verify(token.substring(4), 'secret_key');

    const user = await UserModel.findOne({ _id: (decodedToken).id });
    return {
      user,
    };
  } catch (err) {
    return { user: null };
  }
}


export function generateToken(user) {
  return `JWT ${jwt.sign({ id: user._id }, 'secret_key')}`;
}