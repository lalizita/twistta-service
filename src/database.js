import mongoose from 'mongoose';

export default function connectDatabase(URI = 'mongodb://localhost/twistta-service', options) {
  return new Promise((resolve, reject) => {
    mongoose.Promise = global.Promise;
    mongoose.connection
      .on('error', error => reject(error))
      .on('close', () => console.log('Database connection closed.'))
      .once('open', () => resolve(mongoose.connections[0]));

    mongoose.connect(URI, { useNewUrlParser: true, ...options });
  });
}