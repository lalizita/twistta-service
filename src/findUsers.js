import UserModel from './models/User';

export async function findUsers(){
  const usersList = await UserModel.find();
  return usersList;
}