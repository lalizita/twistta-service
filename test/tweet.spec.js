import { graphql } from 'graphql';
import mongoose from 'mongoose';
import { connectMongoose, createUser, clearDatabase, createTweet } from './helper';
import { schema } from '../src/schema';
const { ObjectId } = mongoose.Types;

describe('tweet test', () => {
  beforeAll(async() => {
    await connectMongoose();
  })
  beforeEach(async() => await clearDatabase())
  it('it should return a tweets list of a user', async() => {
    const text1 = 'Hello People! I wanna show u my graphQL!';
    const text2 = 'Suck my graphQL!';
    const { _id } = await createUser('jon snow', 'jon@stark.com', '543212');
    await createTweet(_id, text1);
    await createTweet(_id, text2);
    global.authorId = ObjectId(_id)
    const query = `
      query Q($id: ID!){
        tweets(id: $id) {
          text
        }
      }
    `
    const variables = {
      id: global.authorId.toString(),
    }

    const { data: { tweets } } = await graphql(schema, query, {}, {}, variables);
    expect(tweets).toMatchSnapshot();
    expect(tweets[0].text).toBe(text1);
    expect(tweets[1].text).toBe(text2);
  })
})