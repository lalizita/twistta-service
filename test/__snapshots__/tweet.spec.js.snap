// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tweet test it should return a tweets list of a user 1`] = `
Array [
  Object {
    "text": "Hello People! I wanna show u my graphQL!",
  },
  Object {
    "text": "Suck my graphQL!",
  },
]
`;
